fairdatapipeline/fdp_utils.py

```diff
@@ -105,7 +104,10 @@ def extract_id(url: str) -> str:
     Returns:
         |   str: id derrived from the url
     """
-    return list(filter(None, urlsplit(url).path.split("/")))[-1]
+    try:
+        return list(filter(None, urlsplit(url).path.split("/")))[-1]
```

Comment:
`list(filter(` does absolutely nothing in this context and is actually confusing, the purpose of this is to simply get the last element of a list, and if the list is empty raise an exception, as `urlsplit(url).path.split("/")` already returns a list that is all that is needed. I do not know why `filter` is being used at all.

Reply:
ok, using your idea.
    split_url_path = urlsplit(url).path.split("/")
    if not split_url_path:
        raise IndexError(f"Unable to extract ID from registry URL: {url}")
    return [s for s in split_url_path if s.strip() != ''][-1]